src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -79,6 +79,9 @@ public class ConsensusManager : IConsensusManager
         /// <inheritdoc />
         public ChainedHeader Tip { get; private set; }
 
+        /// <inheritdoc />
+        public int? HeaderTip => this.chainedHeaderTree.GetBestPeerTip()?.Height ?? this.Tip.Height;
```

Comment:
What is incorrect in this change?

Reply:
```
/// <remarks>All access should be protected by <see cref="peerLock"/>.</remarks>
private readonly IChainedHeaderTree chainedHeaderTree;
```

It would be advisable if someone with more knowledge of ConsensusManager be included in review when making changes in that area